lib/util/mixins.js

```diff
@@ -80,32 +77,15 @@ module.exports = {
 		}
 
 		const parts = this.extract( options.messageType );
-		let returnData;
 
 		if ( parts instanceof Error ) {
-			returnData = parts;
-		} else {
-			// if no exchange, create exchange from msgType
-			const partExchange = options.exchange ||  parts.resource;
-			const exchange     = prefix + '.' + partExchange + '-exchange';
-
-			// if no queue, create queue from msgType
-			const partQueue = options.queue || parts.action;
-			const queue     = prefix + '.' + parts.resource + '-' + partQueue;
-
-			// should we override the exchange and messageType
-			const transformedOptions = {
-				queue,
-				exchange,
-				'messageType' : prefix + '.' + options.messageType
-			};
-
-			// strip off to persists transformed options
-			options = _.omit( options,
-								'messageType', 'exchange', 'queue', 'validate', 'validateOptions' );
-
-			returnData = _.extend( transformedOptions, options );
+			return parts;
 		}
-		return returnData;
+
+		return Object.assign( {}, options, {
+			'queue'       : options.queue || `${prefix}.${parts.resource}-${parts.action}`,
```

Comment:
This is different than was previously in there. I'm not sure if you meant to do this. I like this way better, because if they pass in an `queue` we shouldn't be prefixing and suffixing it, but the old logic did.